src/citrine/informatics/workflows/design_workflow.py

```diff
@@ -50,20 +52,54 @@ def __init__(self,
                  design_space_id: Optional[UUID] = None,
                  predictor_id: Optional[UUID] = None,
                  predictor_version: Optional[Union[int, str]] = None,
+                 data_source: Optional[DataSource] = None,
                  description: Optional[str] = None):
         self.name = name
         self.design_space_id = design_space_id
         self.predictor_id = predictor_id
         self.predictor_version = predictor_version
+        self.data_source = data_source
         self.description = description
 
     def __str__(self):
         return '<DesignWorkflow {!r}>'.format(self.name)
 
+    @classmethod
+    def _pre_build(cls, data: dict) -> dict:
+        """Run data modification before building."""
+        data_source_id = data.pop("data_source_id", None)
+        if data_source_id is not None:
+            data["data_source"] = DataSource.from_data_source_id(data_source_id).dump()
+        return data
+
+    def _post_dump(self, data: dict) -> dict:
+        """Run data modification after dumping."""
+        data_source = data.pop("data_source", None)
+        if data_source is not None:
+            data["data_source_id"] = DataSource.build(data_source).to_data_source_id()
+        else:
+            data["data_source_id"] = None
+        return data
+
```

Comment:
This uses the existing serde hooks to filter out `data_source_id` fields and insert DataSource objects.